src/inet/TCPEndPointImplLwIP.cpp

```diff
@@ -653,53 +653,56 @@ void TCPEndPointImplLwIP::HandleDataSent(uint16_t lenSent)
 void TCPEndPointImplLwIP::HandleDataReceived(System::PacketBufferHandle && buf)
 {
     // Only receive new data while in the Connected or SendShutdown states.
-    if (mState == State::kConnected || mState == State::kSendShutdown)
+    if (mState != State::kConnected && mState != State::kSendShutdown)
     {
-        // Mark the connection as being active.
-        MarkActive();
+        return;
```

Comment:
The diff here is terrible... the code was changed to early return instead of wrapping everything in an `if {}` but remains the same